app/code/Magento/Customer/Model/AccountManagement.php

```diff
@@ -1588,6 +1595,25 @@ public function getPasswordHash($password)
         return $this->encryptor->getHash($password, true);
     }
 
+    /**
+     * Retrieve and validate existing password reset token; returns true on
+     * success, false on failure
+     *
+     * @param CustomerInterface $customer
+     * @return bool
+     */
+    protected function customerHasValidPasswordResetToken(CustomerInterface $customer): bool
+    {
+        try {
+            $customerId = $customer->getId();
+            $customerSecureData = $this->customerRegistry->retrieveSecureData($customerId);
+            $token = $customerSecureData->getRpToken();
+            return $this->validateResetPasswordToken($customerId, $token);
+        } catch (\Exception $exception) {
```

Comment:
I think we're not expecting to to have there for instance "unable to connect to DB" exceptions, right? So we shouldn't catch there too general exceptions.